### PR TITLE
feat(auth): Add `TotpInfo` field to `UserRecord`

### PR DESCRIPTION
**Pull request by pragatimodi**
Adding a `TotpInfo` field to `UserRecord` for Admin users to view a user's MFA configuration. TOTP enrollment for users is enabled via client SDKs only.
Testing was done by manually enrolling a user on TOTP using curl and creating a test app to get the corresponding user's UserRecord configuration
Manual testing output: 
```
{
  "MultiFactor": {
    "EnrolledFactors": [
      {
        "UID": "d3dc0070-665b-42e1-b3ec-18f98af224fb",
        "DisplayName": "totp-enroll-test",
        "EnrollmentTimestamp": 1685671931000,
        "FactorID": "totp",
        "PhoneMultiFactorInfo": null,
        "TOTPMultiFactorInfo": {}
      }
    ]
  }
}
```

### CODE REVIEW

File: auth/user_mgt.go
Comment (renkelvin) [inline auth/user_mgt.go:92]: Change to TOTPMultiFactorInfo to consist with PhoneMultiFactorInfo?

File: auth/user_mgt.go
Comment (renkelvin) [inline auth/user_mgt.go:359]: Add more description about this link?

File: testdata/get_user.json
Comment (renkelvin) [inline testdata/get_user.json:38]: enrolledPhoneFactor? and elsewhere

**Review by renkelvin — APPROVED** [review]
LGTM with nits.

File: auth/user_mgt.go
Comment (pragatimodi) [reply auth/user_mgt.go:92]: It is called `TOTPInfo` in `authFactorInfo` which is what the server definition looks like. For consistency alone, do you think it is worth creating another empty struct?

Comment (renkelvin) [reply auth/user_mgt.go:92]: Yeah, I think so. Including "MultiFactor" is necessary to me.

**Review by Xiaoshouzi-gh — APPROVED** [review]

File: auth/user_mgt.go
Comment (pragatimodi) [reply auth/user_mgt.go:94]: Yes

File: auth/user_mgt_test.go
Comment (pragatimodi) [reply auth/user_mgt_test.go:72]: I would prefer it this way since we can test UIDs with integration tests but for more clarity as we add unit tests this might be better. What do you think? We can revert back to the previous UID and add comments if you feel strongly about this.

File: auth/user_mgt.go
Comment (pragatimodi) [reply auth/user_mgt.go]: Looks like it isn't. Linking the approved PR here for more context on this: https://github.com/firebase/firebase-admin-go/pull/530

File: auth/user_mgt.go
Comment (prameshj) [reply auth/user_mgt.go:94]: So we cannot merge this until the API change is approved, right? Can we keep PhoneNumber for now and merge the rest?

File: auth/user_mgt.go
Comment (prameshj) [reply auth/user_mgt.go:668]: Can we verify this with the REST API?

File: auth/user_mgt_test.go
Comment (prameshj) [reply auth/user_mgt_test.go:72]: > since we can test UIDs with integration tests but for more clarity as we add unit tests this might be better

I am not sure what the human-readable UID gives us that the other alphanumeric value does not. Can you elaborate?

Comment (pragatimodi) [reply auth/user_mgt_test.go:72]: Like you said, human-readable. It is just easier to write unit test cases as we add more factors here. I guess it doesn't make a lot of difference to have alphanumeric strings for unit tests as such because these are generated by the backend in the actual flow.

File: auth/user_mgt.go
Comment (pragatimodi) [reply auth/user_mgt.go:668]: Was able to verify that we don't require `mfaEnrollmentId` while updating `enrollments` in curl.

File: auth/user_mgt.go
Comment (pragatimodi) [reply auth/user_mgt.go:94]: Working on the API proposal for this.

**Comment by pragatimodi** [comment]
Moving this to https://github.com/firebase/firebase-admin-go/pull/573 as per offline sync with @lahirumaramba to base it off `dev` branch. Closing this PR.

**Pull request closed without merging**
